### ISSUE TITLE
Play sound error when multiple players online

### ISSUE THREAD

**Issue by kevin777787** [SkriptLang/Skript#1812]
### Description
I think the play sound effect doesn't work when multiple players are online. It does work when only one player is online.

### Steps to Reproduce
  trigger:
    play sound "announce" with volume 100 at all players for all players

### Expected Behavior
The sound is played to all players without any errors.

### Errors / Screenshots
An internal error occurred when attempting to perform this command.

```
[21:45:40] [Server thread/ERROR]: #!#! Stack trace:
[21:45:40] [Server thread/ERROR]: #!#! ch.njol.skript.SkriptAPIException: Call to getSingle() on a non-single expression
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.util.SimpleExpression.getSingle(SimpleExpression.java:64)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.util.ConvertedExpression.getSingle(ConvertedExpression.java:151)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.effects.EffPlaySound.execute(EffPlaySound.java:81)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.Effect.run(Effect.java:52)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:61)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.lang.Trigger.execute(Trigger.java:57)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.command.ScriptCommand.execute2(ScriptCommand.java:286)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.command.ScriptCommand.execute(ScriptCommand.java:243)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.command.Commands.handleCommand(Commands.java:254)
[21:45:40] [Server thread/ERROR]: #!#!     at ch.njol.skript.command.Commands$1.onPlayerCommand(Commands.java:169)
[21:45:40] [Server thread/ERROR]: #!#!     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[21:45:40] [Server thread/ERROR]: #!#!     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[21:45:40] [Server thread/ERROR]: #!#!     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[21:45:40] [Server thread/ERROR]: #!#!     at java.lang.reflect.Method.invoke(Method.java:498)
[21:45:40] [Server thread/ERROR]: #!#!     at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302)
[21:45:40] [Server thread/ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62)
[21:45:40] [Server thread/ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500)
[21:45:40] [Server thread/ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PlayerConnection.handleCommand(PlayerConnection.java:1391)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1234)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PacketPlayInChat.a(PacketPlayInChat.java:45)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PacketPlayInChat.a(PacketPlayInChat.java:1)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13)
[21:45:40] [Server thread/ERROR]: #!#!     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[21:45:40] [Server thread/ERROR]: #!#!     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:748)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[21:45:40] [Server thread/ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[21:45:40] [Server thread/ERROR]: #!#!     at java.lang.Thread.run(Thread.java:748)
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! Version Information:
[21:45:40] [Server thread/ERROR]: #!#!   Skript: 2.3.1 (latest)
[21:45:40] [Server thread/ERROR]: #!#!     Flavor: skriptlang-github
[21:45:40] [Server thread/ERROR]: #!#!     Date: 2019-01-05
[21:45:40] [Server thread/ERROR]: #!#!   Bukkit: 1.12.2-R0.1-SNAPSHOT
[21:45:40] [Server thread/ERROR]: #!#!   Minecraft: 1.12.2
[21:45:40] [Server thread/ERROR]: #!#!   Java: 1.8.0_191 (OpenJDK 64-Bit Server VM 25.191-b12)
[21:45:40] [Server thread/ERROR]: #!#!   OS: Linux amd64 3.10.0-957.1.3.el7.x86_64
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! Server platform: Spigot
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! Current node: null
[21:45:40] [Server thread/ERROR]: #!#! Current item: play sound
[21:45:40] [Server thread/ERROR]: #!#! Current trigger: command /buyannouncer (simple event) (buyannounce.sk, line -1)
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! Thread: Server thread
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! Language: english
[21:45:40] [Server thread/ERROR]: #!#! Link parse mode: DISABLED
[21:45:40] [Server thread/ERROR]: #!#! 
[21:45:40] [Server thread/ERROR]: #!#! End of Error.
```

### Server Information
* **Server version/platform:**       CraftBukkit version git-Spigot-dcd1643-e60fc34 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
* **Skript version:**                2.3.1

### Additional Context
None.

**Comment by ShaneBeee** [comment]
I could be totally wrong, but here is my theory.
you are doing `at all players` .... where as the expression is `at %location%`
so you are trying to use multiple locations for a single location expression

**Comment by TheBentoBox** [comment]
Shane is correct, this effect only supports one location. If you want to play the effect at multiple people, I'd do it like this (I changed the volume to 2 because that's the max):
```
loop players:
    play sound "announce" with volume 2 at loop-player for loop-player
```
That being said, the effect should be improved to reject syntax which provides multiple locations if possible.

**Comment by Snow-Pyon** [comment]
This is actually a bug, the expression shouldn't be able to take multiple locations if the syntax says so, yet, it is still initing the expression (whereas it should throw an error saying that the effect accepts only a single location, not more).

My guess is that it tries to convert the Player[] obiect to a Location object and somehow passing without erroring, and then failing at execution.

**Comment by Blueyescat** [comment]
bug

**Comment by TheBentoBox** [comment]
This was likely fixed by #1928.
